src/firebase_functions/private/serving.py

```diff
@@ -0,0 +1,114 @@
+"""
+Module used to serve Firebase functions locally and remotely.
+"""
+# pylint: disable=protected-access
+import os
+import inspect
+import enum
+import yaml
+import importlib
+from os import kill, getpid
+from signal import SIGTERM
+
+from flask import Flask
+from flask import Response
+
+from firebase_functions.private import manifest as _manifest
+from firebase_functions.private import util as _util
+
+
+def get_functions():
+    spec = importlib.util.spec_from_file_location("main", "main.py")
+    if spec is not None and spec.loader is not None:
+        module = importlib.util.module_from_spec(spec)
+        spec.loader.exec_module(module)
+    else:
+        raise Exception(
+            "Firebase Functions for Python could not find the main.py file in your project."
+        )
+    functions = inspect.getmembers(module, inspect.isfunction)
+    firebase_functions = {}
+    for entry in functions:
+        if hasattr(entry[1], "__firebase_endpoint__"):
+            name = entry[1].__firebase_endpoint__.entryPoint
+            firebase_functions[name] = entry[1]
+    return firebase_functions
+
+
+def to_spec(data: dict) -> dict:
+
+    def convert_value(obj):
+        if isinstance(obj, enum.Enum):
+            return obj.value
+        if isinstance(obj, dict):
+            return to_spec(obj)
+        if isinstance(obj, list):
+            return list(map(convert_value, obj))
+        return obj
+
+    without_nones = dict(
+        (k, convert_value(v)) for k, v in data.items() if v is not None)
+    return without_nones
+
+
+def functions_as_yaml(functions: dict) -> str:
+    endpoints: dict[str, _manifest.ManifestEndpoint] = {}
+    for name, function in functions.items():
+        endpoint = function.__firebase_endpoint__
+        # v2 function name(s) can only contain lower case letters, numbers, hyphens
+        endpoints[name.replace("_", "-").lower()] = endpoint
+
+    manifest_stack = _manifest.ManifestStack(endpoints=endpoints)
+    manifest_spec = _manifest.manifest_to_spec_dict(manifest_stack)
+    manifest_spec_with_sentinels = to_spec(manifest_spec)
+
+    def represent_sentinel(self, _):
+        # TODO distinguishing between RESET_VALUE or DEFAULT_VALUE
+        # TODO can be done here
```

Comment:
What difference between RESET_VALUE and DEFAULT_VALUE? Sorry, I remember that we had two uses of sentinel but I can't recall what the second case was and those two names sound unfortunately close together.